game/scripts/vscripts/items/azazel_tower_watch.lua

```diff
@@ -0,0 +1,142 @@
+  then
+    return UF_FAIL_CUSTOM
+  else
+    return UF_SUCCESS
+  end
+end
+function item_azazel_tower_watch_1:GetCustomCastErrorLocation(location)
+  return "#dota_hud_error_no_buildings_here"
+end
+
+function item_azazel_tower_watch_1:OnSpellStart()
+  local caster = self:GetCaster()
+  local location = self:GetCursorPosition()
+  local building = CreateUnitByName("npc_azazel_tower_watch", location, true, caster, caster:GetOwner(), caster:GetTeam())
+  building:SetOwner(caster)
+  GridNav:DestroyTreesAroundPoint(location, building:GetHullRadius(), true)
+  building:SetOrigin(location)
+  building:RemoveModifierByName('modifier_invulnerable')
+  building:AddNewModifier(building, self, "modifier_watch_tower_construction", {duration = -1})
+  local charges = self:GetCurrentCharges() - 1
+  if charges < 1 then
+    caster:RemoveItem(self)
+  else
+    self:SetCurrentCharges(charges)
+  end
+end
+
+-- upgrades
+item_azazel_tower_watch_2 = item_azazel_tower_watch_1
+item_azazel_tower_watch_3 = item_azazel_tower_watch_1
+item_azazel_tower_watch_4 = item_azazel_tower_watch_1
+
+--------------------------------------------------------------------------
+-- base modifier
+
+modifier_watch_tower_construction = class(ModifierBaseClass)
+
+local SINK_HEIGHT = 200
+local THINK_INTERVAL = 0.1
+
+function modifier_watch_tower_construction:OnCreated()
+  local ab = self:GetAbility()
+  local level = ab:GetLevel()
+  self.level = level
+  if IsServer() then
+    local target = self:GetParent()
+    local maxhealth = target:GetMaxHealth() + ab:GetSpecialValueFor("bonus_health")
+    local location = target:GetOrigin()
+    local time = ab:GetSpecialValueFor("construction_time")
+    target:Attribute_SetIntValue("construction_time", time)
+    target:Attribute_SetIntValue("bonus_vision_range", ab:GetSpecialValueFor("bonus_vision_range"))
+    target:SetOrigin(GetGroundPosition(location, target) - Vector(0, 0, SINK_HEIGHT))
+    Timers:CreateTimer(0.1, function()
+      ResolveNPCPositions(location, target:GetHullRadius())
+      target:SetMaxHealth(maxhealth)
+      target:SetHealth(maxhealth * 0.01)
+      target:SetMaterialGroup('radiant_level'..level)
+      self:StartIntervalThink(THINK_INTERVAL)
+      self:SetStackCount(math.floor(time / THINK_INTERVAL)) -- `construction_time` should be divisible by `THINK_INTERVAL`!
+    end)
+  end
+end
+
+function modifier_watch_tower_construction:OnIntervalThink()
+  if IsServer() then
+    local target = self:GetParent()
+    local count = self:GetStackCount()
+    if count > 0 then
+      local time = target:Attribute_GetIntValue("construction_time", 10)
+      local location = target:GetOrigin()
+      target:SetOrigin(target:GetOrigin() + Vector(0, 0, SINK_HEIGHT / (time / THINK_INTERVAL)))
+      self:SetStackCount(count - 1)
+    else
+      AddFOWViewer(target:GetTeam(), target:GetOrigin(), target:GetCurrentVisionRange(), THINK_INTERVAL, false)
```

Comment:
This is cringey as anything, but I'm actually not sure there's a better way to do it. I would at least like it to have a higher refresh interval, though I'm unsure what the cleanest way to do that would be. Maybe just another modifier, and have this one self-destruct like the other construction modifiers ought to do.

Reply:
I can make a higher interval for this.